test/integration/test_ibm_job.py

```diff
@@ -138,6 +138,37 @@ def test_retrieve_jobs(self):
         for job in job_list:
             self.assertTrue(isinstance(job.job_id(), str))
 
+    def test_retrieve_legacy_jobs(self):
+        """Test retrieving legacy jobs."""
+        job_list = self.provider.backend.jobs(
+            backend_name=self.sim_backend.name,
+            status="DONE",
+            limit=5,
+            skip=0,
+            legacy=True,
+        )
+        # instance used may not have any legacy jobs
```

Comment:
Why not used the old provider to submit a job to guarantee at least one is returned? Otherwise we can't really tell if the code works.

Reply:
Because then we would have to add back the endpoint for running jobs from the old provider 😅 - but I agree, that would be the proper way to test, so i'll add it back

Reply:
oh I meant just install qiskit-ibmq-provider in the test env and use that to submit jobs